pkg/pipeline/kafka_end_to_end_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package pipeline
+
+import (
+	"testing"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/test"
+	kafkago "github.com/segmentio/kafka-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+const kafkaConfigTemplate = `---
+log-level: debug
+pipeline:
+  - name: kafka_ingest
+  - name: decode_json
+    follows: kafka_ingest
+  - name: transform_generic
+    follows: decode_json
+  - name: kafka_encode
+    follows: transform_generic
+parameters:
+  - name: kafka_ingest
+    ingest:
+      type: kafka
+      kafka:
+        brokers: ["localhost:9092"]
+        topic: topic_in
+        groupid: group_test_in
+  - name: decode_json
+    decode:
+      type: json
+  - name: transform_generic
+    transform:
+      type: generic
+      generic:
+        rules:
+          - input: Bytes
+            output: k_bytes
+          - input: DstAddr
+            output: k_dstAddr
+          - input: DstPort
+            output: k_dstPort
+          - input: Packets
+            output: k_packets
+          - input: SrcAddr
+            output: k_srcAddr
+          - input: SrcPort
+            output: k_srcPort
+  - name: kafka_encode
+    encode:
+      type: kafka
+      kafka:
+        address: "localhost:9092"
+        topic: topic_out
+`
+
+func Test_KafkaPipeline(t *testing.T) {
+	var mainPipeline *Pipeline
+	var err error
+	v := test.InitConfig(t, kafkaConfigTemplate)
+	require.NotNil(t, v)
+
+	mainPipeline, err = NewPipeline()
+	require.NoError(t, err)
+
+	// run the pipeline in a separate go-routine
+	go func() {
+		mainPipeline.Run()
+	}()
+
+	// wait for the pipeline to come up
+	time.Sleep(time.Second)
+
+	inputData := [][]byte{
```

Comment:
Any chance that you change this to feed the log lines from one of the files ??? 

Or at least generalize this so that we will be able to use this example set of inputs also for other end2end tests ?

Reply:
done